pkg/controller/external_tfpluginsdk.go

```diff
@@ -443,7 +443,16 @@ func (n *terraformPluginSDKExternal) getResourceDataDiff(tr resource.Terraformed
 		instanceDiff.RawPlan = v
 	}
 	if instanceDiff != nil && !instanceDiff.Empty() {
-		n.logger.Debug("Diff detected", "instanceDiff", instanceDiff.GoString())
+		encodedDiff, err := json.TFParser.Marshal(map[string]any{
+			"attributes":     instanceDiff.Attributes,
+			"destroy":        instanceDiff.Destroy,
+			"destroyTainted": instanceDiff.DestroyTainted,
+			"destroyDeposed": instanceDiff.DestroyDeposed,
+		})
+		if err != nil {
+			n.logger.Debug("Diff detected", "instanceDiff", instanceDiff.GoString())
+		}
+		n.logger.Debug("Diff detected", "instanceDiff", string(encodedDiff))
```

Comment:
I think this needs to be in an else clause, because right now it will be executed even if there's a deserialization error